### ISSUE TITLE
failed tests missing from tests.obspy.org

### ISSUE THREAD

**Issue by ThomasLecocq** [obspy/obspy#2953]
https://github.com/obspy/obspy/actions/runs/1709170752 has failed tests, but on https://tests.obspy.org/?pr=2662 they don't appear

**Comment by trichter** [comment]
Strange, the test report is uploaded https://tests.obspy.org/117184/

But the tests list is completely empty in the reported JSON.

@d-chambers Do you have a clue what is going on?

Maybe a bug in pytest-json-report?

**Comment by d-chambers** [comment]
> Maybe a bug in pytest-json-report?

No,

It looks like pytest didn't run anything because there was an error when collecting the tests: 

```python
/usr/share/miniconda3/envs/test3.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
core/tests/test_network.py:25: in <module>
    from obspy.imaging.maps import HAS_BASEMAP
E   ImportError: cannot import name 'HAS_BASEMAP' from 'obspy.imaging.maps' (/home/runner/work/obspy/obspy/obspy/imaging/maps.py)
```

If one of the test files is not importable the test collection will fail. Curious the report was still generated and uploaded though.

**Comment by ThomasLecocq** [comment]
I knew I'd find a way to break your system :-)

**Comment by d-chambers** [comment]
I never doubted you for a moment 😝 

I don't think we really need to make any changes per se. An empty test report means there is an error in importing one of the test files.

Or, I guess pytest might emit a different return code if the collection fails and we can raise a big error message to avoid uploading and empty report, but it probably ins't much informative than what it is already doing.

**Comment by QuLogic** [comment]
> Or, I guess pytest might emit a different return code if the collection fails and we can raise a big error message to avoid uploading and empty report, but it probably ins't much informative than what it is already doing.

It might return a 5? https://docs.pytest.org/en/latest/reference/exit-codes.html

**Comment by trichter** [comment]
> https://docs.pytest.org/en/latest/reference/exit-codes.html

Interesting, but according to the uploaded json, the exit code had been 2.
We could check for exit code > 1 and not upload any test report in this case.

**Comment by d-chambers** [comment]
closed by #2955
